File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/Linux/X64/core/handler/abstract_command_handler.rb
require_relative 'handler_dictionary'

class AbstractCommandHandler

  @required_parameters_count = 0

  def handle_command(command)
    iterate(command)
    return process(command)
  end

  def iterate(command)
    (0..command.payload.length).step(1) do |i|
      if command.payload[i].is_a? Command
        command.payload[i] = $handler_dict[command.payload[i].command_type].handle_command(command.payload[i])
      end
    end
  end

  def process(command)
    raise 'process is not implemented'
  end

end

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/handler/AbstractCommandHandler.py
from javonet.core.handler.HandlerDictionary import handler_dict
from javonet.utils.Command import Command


class AbstractCommandHandler:
    _required_parameters_count = 0

    def process(self, command):
        pass

    def handle_command(self, command):
        self.__iterate(command)
        return self.process(command)

    @staticmethod
    def __iterate(command):
        for i in range(0, len(command.payload)):
            if isinstance(command.payload[i], Command):
                command.payload[i] = handler_dict.get(command.payload[i].command_type).handle_command(
                        command.payload[i])
